demisto_sdk/commands/validate/tests/IN_validators_test.py

```diff
@@ -4376,6 +4382,42 @@ def test_IsHiddenableParamValidator_is_valid(
     )
 
 
+def test_IsHiddenableParamValidator_with_old_content_object_is_valid():
+    """
+    Given
+    content_items iterables.
+        - Case 1: Four invalid integrations with old content objects with different git statuses:
+            - One integration with a non-hiddenable param with hidden value = True with modified status.
+            - One integration with a non-hiddenable param with hidden value = True and type not in 0,4,12,14, with a type 9 replacement with renamed status.
+            - One integration with a non-hiddenable param with hidden value = 'true' with added status.
+            - One integration with a non-hiddenable param with hidden value = all market places and another hidden type 4 param without type 9 replacement with modified status.
+    When
+    - Calling the IsHiddenableParamValidator is valid function.
+    Then
+        - Make sure the validation fail when it needs to and the right error message is returned.
+        - Case 1: Should pass all except for the third content item.
+    """
+    content_items = INVALID_HIDDEN_PARAM_INTEGRATIONS
+    old_content_items = copy.deepcopy(content_items)
+    for old_content_item, git_status in zip(
+        old_content_items,
+        [
+            GitStatuses.MODIFIED,
+            GitStatuses.RENAMED,
+            GitStatuses.ADDED,
+            GitStatuses.MODIFIED,
+        ],
+    ):
+        old_content_item.git_status = git_status
+    create_old_file_pointers(content_items, old_content_items)
+    results = IsHiddenableParamValidator().is_valid(content_items)
+    assert len(results) == 1
```

Comment:
lets add test cases for the flow it shouldnt fail because it was already hidden

Reply:
That's exactly what the new test does.